microcosm_postgres/tests/test_company_store.py

```diff
@@ -93,6 +93,22 @@ def test_create_delete_company(self):
             raises(ModelNotFoundError, pattern="Company not found"),
         )
 
+    def test_create_delete_company_complicated_expression(self):
+        """
+        Delete should support more complicated criterion with the `fetch` synchronization strategy enabled.
+
+        """
+        with transaction():
+            company = Company(name="name").create()
```

Comment:
Just for sanity, can you create another company here?

Reply:
Done!